Upstream: No, see https://github.com/sysown/proxysql/pull/3401
From afebf510872af492be9daf52c3bb5582604a1562 Mon Sep 17 00:00:00 2001
From: Arnaud Lefebvre <a.lefebvre@outlook.fr>
Date: Fri, 9 Apr 2021 19:30:00 +0200
Subject: [PATCH 2/2] libinjection: use CC, LD and AR from environment if
 available

This commit uses a patch sent to
https://github.com/libinjection/libinjection/pull/15
---
 deps/Makefile                             |  1 +
 deps/libinjection/cc-ar-ld-from-env.patch | 38 +++++++++++++++++++++++
 2 files changed, 39 insertions(+)
 create mode 100644 deps/libinjection/cc-ar-ld-from-env.patch

diff --git a/deps/Makefile b/deps/Makefile
index 149f4f79..07bbddc0 100644
--- a/deps/Makefile
+++ b/deps/Makefile
@@ -72,6 +72,7 @@ libinjection/libinjection/src/libinjection.a:
 ifneq ($(CENTOSVER),6)
 	cd libinjection/libinjection && patch -p1 < ../update-build-py3.diff
 	cd libinjection/libinjection && patch -p1 < ../libinjection_sqli.c.patch
+	cd libinjection/libinjection && patch -p1 < ../cc-ar-ld-from-env.patch
 endif
 ifeq ($(UNAME_S),Darwin)
 	sed -i '' 's/CC=/CC?=/' libinjection/libinjection/src/Makefile
 
diff --git a/deps/libinjection/cc-ar-ld-from-env.patch b/deps/libinjection/cc-ar-ld-from-env.patch
new file mode 100644
index 00000000..2242b47e
--- /dev/null
+++ b/deps/libinjection/cc-ar-ld-from-env.patch
@@ -0,0 +1,38 @@
+Upstream: Yes: https://github.com/libinjection/libinjection/pull/15
+From 3193a70beb53ff6a39673b47002f6fe4165110a8 Mon Sep 17 00:00:00 2001
+From: Arnaud Lefebvre <a.lefebvre@outlook.fr>
+Date: Fri, 9 Apr 2021 19:16:56 +0200
+Subject: [PATCH] Makefile: use CC, LD and AR from environment if available
+
+---
+ src/Makefile | 8 ++++----
+ 1 file changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/src/Makefile b/src/Makefile
+index f288176..48bdc6a 100644
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -1,10 +1,10 @@
+ 
+-AR=ar -r
++AR ?= ar
+ # MAC
+ # AR=libtool -static
+ 
+-CC=cc
+-LD=ld
++CC ?= cc
++LD ?= ld
+ CFLAGS=-Wall -Wextra -Werror -pedantic -ansi -g -O3 -fPIC
+ 
+ INSTALL=install
+@@ -67,7 +67,7 @@ ${SHAREDLIB}: ${OBJECTS}
+ 
+ ${STATICLIB}: ${OBJECTS}
+ 	rm -f ${STATICLIB}
+-	${AR} ${STATICLIB} ${OBJECTS}
++	${AR} -r ${STATICLIB} ${OBJECTS}
+ 
+ check: ${TESTDRIVERS}
+ 	@./test-driver.sh test-unit.sh
+
-- 
2.31.1

